File: PAT-Basic/B1002.cpp
#include<stdio.h>
#include<string.h>

char pin[10][5] = {"ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu"};
int main() {
	char str[110];
	int num = 0, len, x[4] = {0}, j;
//	scanf("%s", str);
	gets(str);
	len = strlen(str);
	for(int i = 0; i < len; i++) {
		num += str[i] - '0';
	}
	for(j = 0; num != 0; j++) {
		x[j] = num % 10;
		num = num/10;
	}
	for(int i = j-1; i >= 0; i--) {
		printf("%s", pin[x[i]]);
		if(i != 0) printf(" ");
	}
	return 0;
}

File: Java/PAT-Basic/B1002.java
import java.util.Scanner;

public class Main {

    static int maxn = 110;
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String[] py = new String[10];
        py[0] = "ling"; py[1] = "yi"; py[2] = "er"; py[3] = "san"; py[4] = "si";
        py[5] = "wu"; py[6] = "liu"; py[7] = "qi"; py[8] = "ba"; py[9] = "jiu";
        String str;
        str = in.nextLine();
        int a = 0;
        for(int i = 0; i < str.length(); i++) {
            a += str.charAt(i) - '0';
        }
        int[] array = new int[10];
        int j = 0, k = 0;
        while (a != 0) {
            array[j++] = a % 10;
            a = a / 10;
        }
        for (int i = j - 1; i >= 0; i--) {
            System.out.print(py[array[i]]);
            if(i != 0) {
                System.out.printf(" ");
            }
        }
    }

}